tools/src/main/scala/org/scalanative/bindgen/Bindgen.scala

```diff
@@ -0,0 +1,164 @@
+package org.scalanative.bindgen
+
+import java.io.File
+
+import scala.collection.mutable
+import scala.sys.process.Process
+
+sealed trait Bindgen {
+
+  /**
+   * Set bindgen executable
+   */
+  def bindgenExecutable(executable: File): Bindgen
+
+  /**
+   * Set header file for which bindings will be generated
+   */
+  def header(header: File): Bindgen
+
+  /**
+   * Library to link with, e.g. -luv for libuv
+   */
+  def link(library: String): Bindgen
+
+  /**
+   * Name of Scala object that contains bindings.
+   * Default is set to library name.
+   */
+  def scalaObjectName(scalaObjectName: String): Bindgen
+
+  /**
+   * Package name of generated Scala file
+   */
+  def packageName(packageName: String): Bindgen
+
+  /**
+   * Declarations will be removed if their names
+   * contain given prefix
+   */
+  def excludePrefix(prefix: String): Bindgen
+
+  /**
+   * Additional argument to append to the compiler command line
+   */
+  def extraArg(arg: String): Bindgen
+
+  /**
+   * Additional argument to append to the compiler command line
+   */
+  def extraArgBefore(arg: String): Bindgen
+
+  /**
+   * Run binding generator
+   */
+  def generate(): Bindings
+}
+
+object Bindgen {
+  def apply(): Bindgen = Impl()
+
+  private final case class Impl() extends Bindgen {
+    private var executable: File                    = _
+    private var library: String                     = _
+    private var header: File                        = _
+    private var scalaObjectName: String             = _
+    private var packageName: String                 = _
+    private var excludePrefix: String               = _
+    private val extraArg: mutable.Seq[String]       = mutable.Seq()
+    private val extraArgBefore: mutable.Seq[String] = mutable.Seq()
```

Comment:
I'd much prefer to move these to the case class members:

```scala
private final case class Impl(
  execuable: File,
  library: String,
  // ...
  extraArgs: immutable.Seq[String],
  // ...
) extends Bindgen {
  def bindgenExecutable((executable: File): Bindgen {
     require(executable.exists())
     copy(executable = executable)
  }
  def extraArgs(args: String*): Bindgen = {
    require(args.forall(_.nonEmpty))
    copy(extraArgs = extraArgs ++ args)
  }
  // ...
}
```
This also allows to use `immutable.Seq` or `List`.